### PR TITLE
Add a test for https://devdiv.visualstudio.com/DevDiv/_workitems?id=278481.

### PR DESCRIPTION
**Pull request by AlekseyTs**
Also added some checks that should help to figure out how to reproduce the crash.

@dotnet/roslyn-compiler Please review.

### CODE REVIEW

File: src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs [hunk 1]
```diff
@@ -487,6 +487,21 @@ private BoundLambda ReallyBind(NamedTypeSymbol delegateType)
                 SourceMemberMethodSymbol.ReportAsyncParameterErrors(lambdaSymbol.Parameters, diagnostics, lambdaSymbol.Locations[0]);
             }
 
+            // This is an attempt to get a repro for https://devdiv.visualstudio.com/DevDiv/_workitems/edit/278481?fullScreen=false
```
Comment (gafter) [inline src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs]: You can probably drop the `?fullScreen=false`

File: src/Compilers/CSharp/Test/Semantic/Semantics/LambdaTests.cs [hunk 2]
```diff
@@ -2290,5 +2290,70 @@ public static class C
             var comp = CreateCompilationWithMscorlib(source);
             CompileAndVerify(comp);
         }
+
+        [Fact, WorkItem(278481, "https://devdiv.visualstudio.com/DevDiv/_workitems?id=278481&fullScreen=false&_a=edit")]
```
Comment (gafter) [inline src/Compilers/CSharp/Test/Semantic/Semantics/LambdaTests.cs]: I recommend a URI of the form `https://devdiv.visualstudio.com/DevDiv/_workitems?id=278481`

Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Semantic/Semantics/LambdaTests.cs]: Will change.

**Comment by AlekseyTs** [comment]
@MattGertz For ask mode approval. 
This change doesn't fix the crash (https://devdiv.visualstudio.com/DevDiv/_workitems?id=278481), but it adds some code that should allow us to get a better idea about what is causing it if it happens again.

**Comment by AlekseyTs** [comment]
@dotnet-bot Retest this please.

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review. I need a second sign-off because the change includes two small code changes.

**Comment by cston** [comment]
LGTM

**Pull request merged**